### PR TITLE
Improve the template tags.

### PR DESCRIPTION
**Pull request by jezdez**

### CODE REVIEW

File: waffle/templatetags/waffle_tags.py [hunk 1]
```diff
@@ -23,72 +25,50 @@ def __iter__(self):
...
-    try:
-        tag, switch_name = token.contents.split(None, 1)
-    except ValueError:
-        raise template.TemplateSyntaxError, \
-              "%r tag requires an argument" % token.contents.split()[0]
-
-    switch_name = switch_name.strip('\'"')
-    condition = lambda r, n: switch_is_active(n)
+def flag(parser, token):
+    flag_is_active, sample_is_active, switch_is_active
```
Comment (jsocol) [inline waffle/templatetags/waffle_tags.py]: This line is a noop, right? Just double checking, I can clean it up after I merge.

Comment (jezdez) [inline waffle/templatetags/waffle_tags.py]: Huh, what the heck? Just a copy/paste error :)

File: waffle/templatetags/waffle_tags.py [hunk 2]
```diff
@@ -23,72 +25,64 @@ def __iter__(self):
...
-    nodelist_true = parser.parse(('else', 'endswitch'))
-    token = parser.next_token()
-    if token.contents == 'else':
-        nodelist_false = parser.parse(('endswitch',))
-        parser.delete_first_token()
-    else:
-        nodelist_false = template.NodeList()
 
-    return WaffleNode(nodelist_true, nodelist_false, condition, switch_name)
+@register.filter('flag')
```
Comment (jsocol) [inline waffle/templatetags/waffle_tags.py:76]: Can you add tests for these filters? I'm not sure how they're supposed to work.

**Comment by jsocol** [comment]
Hey Jannis, I'd love to merge this but I'm still a bit unclear on the new filters, any chance you could add simple tests for them?

**Comment by jezdez** [comment]
Whoops! Sorry @jsocol must have missed your previous comment, will work on some docs and tests, sorry :)

**Comment by jsocol** [comment]
Thanks! I'm actually mucking about with the docs right now, so no rush on that part.

**Comment by jezdez** [comment]
Now that I thought about this "improvement" again, I think I disagree with myself :) I needed the result of a flag check in a template context variable and thought that writing `{% with feature_enabled="some_feature"|flag:request %}..{% endwith %}` is more readable than passing in the value from the view. I'm closing this ticket since I was obviously not right.

**Pull request closed without merging**